DEVINFO.md

```diff
@@ -0,0 +1,174 @@
+
+Therefore, making changes to the codebase is necessarily a review-intensive process.  No one wants bugs, but **no one can afford consensus bugs**.  This page describes how to make and review _non-consensus_ changes.  The process for consensus changes includes not only the entirety of this document, but also the [SIP process](https://github.com/stacksgov/sips/blob/main/sips/sip-000/sip-000-stacks-improvement-proposal-process.md).
+
+A good PR review sets both the submitter and reviewers up for success.  It minimizes the time required by both parties to get the code into an acceptable state, without sacrificing quality or safety.  Unlike most other software development practices, _safety_ is the primary concern.  A PR can and will be delayed or closed if there is any concern that it will lead to unintended consensus-breaking changes.
+
+This document describes some best practices on how to create and review PRs in this context.  The target audience is people who have commit access to this repository (reviewers), and people who open PRs (submitters).  This is a living document -- developers can and should document their own additional guidelines here.
+
+This document is formatted like a checklist.  Each paragraph is one goal or action item that the reviewer and/or submitter must complete.  The **key take-away** from each paragraph is bolded.
+
+## Reviewer Expectations
+
+The overall task of a reviewer is to create an **acceptance plan** for the submitter.  This is simply the list of things that the submitter _must_ do in order for the PR to be merged.  The acceptance plan should be coherent, cohesive, succinct, and complete enough that the reviewer will understand exactly what they need to do to make the PR worthy of merging, without further reviews.  The _lack of ambiguity_ is the most important trait of an acceptance plan.
+
+Reviewers should **strive to complete the review in one round**.  The reviewer should provide enough detail to the submitter that the submitter can make all of the requested changes without further supervision.  Whenever possible, the reviewer should provide all of these details publicly as comments, so that _other_ reviewers can vet them as well.  If a reviewer _cannot_ complete the review in one round due to its size and complexity, then the reviewer may request that the PR be simplified or broken into multiple PRs.
```

Comment:
When tagging two reviewers, should they work in parallel or sequentially?

Reply:
Doesn't matter.  What is important is that no one "freeloads" off of another -- you should review the code as if you're the last line of defense against shipping a show-stopping bug.

Reply:
Is there a proposal for what an "acceptance plan" is?
depending on who's reviewing a PR, that plan may differ greatly from PR to PR unless that is assumed and by design?

Reply:
An acceptance plan is just the (informal) list of things the reviewer expects the submitter to do in order to get their PR merged.  Usually, this is just the act of addressing all comments.  Sometimes, it can be more in-depth, such as changing the design or factoring or some such.  I left the description unspecified because it'll likely be specific to both the PR and the people involved in getting it ready.